ui/lib/core/addon/components/search-select.js

```diff
@@ -1,99 +1,123 @@
- * see the power-select docs for more information.
+ // * advanced customization
+ * @param {Array} options - array of objects passed directly to the power-select component. If doing this, `models` should not also be passed as that will overwrite the
+ * passed options. ex: [{ name: 'namespace45', id: 'displayedName' }];
+ * @param {function} search - Customizes how the power-select component searches for matches - see the power-select docs for more information.
  *
  */
-export default Component.extend({
-  layout,
-  'data-test-component': 'search-select',
-  attributeBindings: ['data-test-component'],
-  classNameBindings: ['displayInherit:display-inherit'],
-  classNames: ['field', 'search-select'],
-  store: service(),
-  onChange: () => {},
-  inputValue: computed(function () {
-    return [];
-  }),
-  allOptions: null, // list of options including matched
-  selectedOptions: null, // list of selected options
-  options: null, // all possible options
-  shouldUseFallback: false,
-  shouldRenderName: false,
-  disallowNewItems: false,
-  passObject: false,
-  objectKeys: null,
-  idKey: computed('objectKeys', function () {
-    // if objectKeys exists, then use the first element of the array as the identifier
-    return this.objectKeys ? this.objectKeys[0] : 'id';
-  }),
-  init() {
-    this._super(...arguments);
-    this.set('selectedOptions', this.inputValue || []);
-  },
-  didRender() {
-    this._super(...arguments);
-    let { oldOptions, options, selectedOptions } = this;
-    let hasFormattedInput = typeof selectedOptions.firstObject !== 'string';
-    if (options && !oldOptions && !hasFormattedInput) {
-      // this is the first time they've been set, so we need to format them
-      this.formatOptions(options);
+
+export default class SearchSelect extends Component {
+  allOptions; // ALL options initially returned from query
+  @service store;
+  @tracked selectedOptions; // array of selected options
+  @tracked dropdownOptions; // options that will render in dropdown
+  @tracked shouldUseFallback = false;
+
+  get disallowNewItems() {
+    return this.args.disallowNewItems || false;
+  }
+
+  get hidePowerSelect() {
+    return this.selectedOptions.length >= this.args.selectLimit;
+  }
+
+  get idKey() {
+    if (this.args.objectKeys) {
+      assert('@objectKeys passed to <SearchSelect> must be an array', Array.isArray(this.args.objectKeys));
     }
-    this.set('oldOptions', options);
-  },
-  formatOptions: function (options) {
-    options = options.toArray().map((option) => {
-      option.searchText = `${option.name} ${option[this.idKey]}`;
+    // if objectKeys exists, use the first element of the array as the identifier
+    return this.args.objectKeys ? this.args.objectKeys[0] : 'id';
+  }
+
+  get passObject() {
+    return this.args.passObject || false;
+  }
+
+  get resetSelection() {
+    return this.args.resetSelection || false;
+  }
```

Comment:
I think unless we need to check for strict equality of `false` these getters are unnecessary.